### ISSUE TITLE
`tf.profiler.profile` outputs negative number of flops

### ISSUE THREAD

**Issue by voegtlel** [tensorflow/tensorflow#20275]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: N/A
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.8.0
- **Python version**: 3.6.5
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: See below

### Describe the problem
`tf.profiler.profile` outputs negative number of flops for `tf.nn.conv2d` ops on large inputs because of np.int32 overflow.

### Source code / logs
```py
import tensorflow as tf
from tensorflow.python.framework.ops import get_stats_for_node_def

test_conv2d = tf.nn.conv2d(tf.zeros((1, 1024, 1024, 16)), tf.zeros((4, 4, 16, 4)), strides=[1, 1, 1, 1], padding='SAME')
# This function is used by tf.profiler.profile
stats = get_stats_for_node_def(tf.get_default_graph(), test_conv2d.op.node_def, 'flops')
# Compare the results: one is negative, the other is the actual result
# The size is exactly 2^31
print(stats.value, 1 * 1024 * 1024 * 4 * 16 * 4 * 4 * 2)
```

The reason is:
https://github.com/tensorflow/tensorflow/blob/cfebbbc94f3edd1622a9a42379dd2ccc956ea52c/tensorflow/python/ops/nn_ops.py#L2169
here `np.prod` is used which returns an `np.int32`. This enforces the next product line 
https://github.com/tensorflow/tensorflow/blob/cfebbbc94f3edd1622a9a42379dd2ccc956ea52c/tensorflow/python/ops/nn_ops.py#L2172
to compute everything as `np.int32`. Actually, the protobuf supports int64:
https://github.com/tensorflow/tensorflow/blob/cf375f06747b7be998f0df329772390f545577a1/tensorflow/core/profiler/tfprof_output.proto#L105
so this line could be replaced by:
```py
np.prod(output_shape.as_list(), dtype=np.int64)
```
There are more ops affected by this issue, as they also use `np.prod`.

PS: Sorry that I cannot submit the fix straight away, everything needed to fix this issue is mentioned above.

**Comment by poxvoculi** [comment]
Thanks for the clear problem diagnosis.  I'll push a fix internally.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @poxvoculi: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @poxvoculi: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
